SleepEngine/Engine/Components/Renderer.h

```diff
@@ -0,0 +1,35 @@
+// Copyright 2019 Tihran Katolikian
+
+#pragma once
+
+BeginNamespaceSleep
+
+class Renderer : public Component
+{
+public:
+    using Base = Component;
+
+    Renderer() noexcept(true);
+
+    void update(float dt) override;
+
+    Color getColor() const { return m_color; }
+    void setColor(Color color) { m_color = color; }
+
+    glm::vec2 getTopLeftUV() const { return m_topLeftUV; }
+    void setTopLeftUV(glm::vec2 uv) { m_topLeftUV = uv; }
+
+    glm::vec2 getDownRightUV() const { return m_downRightUV; }
+    void setDownRightUV(glm::vec2 uv) { m_downRightUV = uv; }
+
+    Texture* getTexture() const { return m_texture; }
+    void setTexture(Texture* texture) { m_texture = texture; }
```

Comment:
Let's use RefGetterAndSetter macro for things like this?  In my impl it required 2 arguments - member itself and member name in function (Texture for m_texture)

Reply:
But I don't want this function to return a reference to a pointer:) I want it to be exactly a getter and a setter.